### PR TITLE
Fix trigrams search for words of 5 characters

### PR DESCRIPTION
**Pull request by diox**
Fixes mozilla/addons#14773

### CODE REVIEW

File: src/olympia/search/filters.py [hunk 1]
```diff
@@ -707,7 +707,7 @@ def primary_should_rules(self, search_query, lang):
                             'match': {
                                 'name.trigrams': {
                                     'query': search_query,
-                                    'minimum_should_match': '66%',
+                                    'minimum_should_match': '67%',
```
Comment (diox) [inline src/olympia/search/filters.py:710]: `3*66/100 = 1.98`
`3*67/100 = 2.01`

Since 5 character words like "Adobe" have 3 trigrams, and `minimum_should_match` is always rounded down to the lowest integer, that change is enough to fix the problem and ensure "Dobromat" doesn't match "Adobe".

File: src/olympia/search/tests/test_search_ranking.py [hunk 2]
```diff
@@ -709,7 +709,7 @@ def test_scenario_tabbicat(self):
         self._check_scenario(
             'tabbicat',
             (
-                ['Tabby Cat', 3255],
```
Comment (diox) [inline src/olympia/search/tests/test_search_ranking.py:712]: That was the only result which had its score affected by this change, because it's not matching as well as before. Still high enough to be considered the best result for that query in our test db though.

File: src/olympia/search/tests/test_search_ranking.py [hunk 3]
```diff
@@ -1063,3 +1063,10 @@ def test_scenario_promoted(self):
                 ['Stripy Dog 4', 584],  # not promoted
             ),
         )
+
+    def test_scenario_minimum_should_match_trigrams(self):
```
Comment (KevinMind) [inline src/olympia/search/tests/test_search_ranking.py:1067]: This test scenario is hard to follow. Is it possible to mock the config for 66 and 67 and have an assert for both to verify modifying that value specifically modifies the expected outcome? This test is probably something that will randomly fail in the future due to some other partially related change and it will end up in a PR similar to Tabby Cat.

Comment (diox) [reply src/olympia/search/tests/test_search_ranking.py:1067]: These tests care about scoring so the way we have set them up is to have an entire set of add-ons that is used for each of the tests. That's fairly slow, so that's why it's the same for each, it's only created once for the entire test class.

Ultimately it's expected that changes to scoring/search algorithm will cause updates to multiple tests, that's fine. There is a script provided in this file to do bulk changes when required (In fact I expected to need to do more changes here, but ultimately didn't have to because of how much an edge case it is)

**Review by KevinMind — APPROVED** [review]
LGTM
